src/lib.rs

```diff
@@ -122,7 +146,17 @@ where
 ///
 /// # Example
 ///
-/// uncompress_archive_file(std::io::Read, std::io::Write, &str)?;
+/// # fn main() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
Looks like we are missing a block delimiter here.

Reply:
Good catch. Fixed.